### ISSUE TITLE
Accessibility - icon-only MenuItem fails accessibility checks

### ISSUE THREAD

**Issue by damienrobson-sage** [Sage/carbon#6017]
### Current behaviour

When using an icon-only `MenuItem` entry, the component causes a CRITICAL failure within e.g. AXE devTools if you pass it an `ariaLabel` prop. This is because aria-label attributes cannot be used on a span with no valid role attribute (which is where the data-component="icon" attribute is declared).

We're currently on version 113.0.1 of Carbon but the issue is still present in the latest build (as demonstrated in the storybook link below)

### Expected behaviour

The icon-only approach is accessible and passes any checks ran against it.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/docs/menu--default-story#with-icon

### Carbon Version

113.0.1

### Design Tokens Version

2.48.0

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Anything else we should know?

__from Chris Merrington__

> For those interested: aria-label is intended for use on interactive elements, or elements made to be interactive via other ARIA declarations, when there is no appropriate text visible in the DOM that could be referenced as a label
> [From aria-label guidance here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)
> So best practice is to have visible text and use that as the accessible name but failing that using aria-label is fine.
> It should be used only on interactive elements but it isn’t automatically a fail if used on something else. In this case the aria-label is just in the wrong place and should be added to the <button>

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by nicktitchmarsh** [comment]
#4986 will address this. Closing as duplicate
